bot/exts/utils/snekio.py

```diff
@@ -0,0 +1,64 @@
+"""I/O File protocols for snekbox."""
+from __future__ import annotations
+
+from base64 import b64decode, b64encode
+from dataclasses import dataclass
+from io import BytesIO
+from pathlib import Path
+
+from discord import File
+
+# Note discord upload limit is 8 MB, or 50 MB for lvl 2 boosted servers
+FILE_SIZE_LIMIT = 8 * 1024 * 1024  # 8 MiB
+
+
+def sizeof_fmt(num: int, suffix: str = "B") -> str:
+    """Return a human-readable file size."""
+    for unit in ("", "Ki", "Mi", "Gi", "Ti", "Pi", "Ei", "Zi"):
+        if abs(num) < 1024:
+            return f"{num:3.1f}{unit}{suffix}"
+        num /= 1024
+    return f"{num:.1f}Yi{suffix}"
+
+
+@dataclass
+class FileAttachment:
+    """File Attachment from Snekbox eval."""
+
+    path: str
+    content: bytes
+
+    def __repr__(self) -> str:
+        """Return the content as a string."""
+        content = f"{self.content[:10]}..." if len(self.content) > 10 else self.content
+        return f"FileAttachment(path={self.path!r}, content={content})"
+
+    @classmethod
+    def from_dict(cls, data: dict, size_limit: int = FILE_SIZE_LIMIT) -> FileAttachment:
+        """Create a FileAttachment from a dict response."""
+        size = data.get("size")
+        if (size and size > size_limit) or (len(data["content"]) > size_limit):
+            raise ValueError("File size exceeds limit")
```

Comment:
The [API](https://discord.com/developers/docs/reference#uploading-files) documentation states the limit applies to all files in a request, not to each file. Not sure why the client lets me upload multiple 8 MiB files. I guess a look at the network request would answer that.

Furthermore, I recall that 4 years ago when I last dealt with this, the limit was on the request size, not just the attached file size. This means space would have to be left for the other request data (like the text message, request headers, etc). Not sure if this is still the case.

Reply:
Added a 10 file upload limit and associated error messages in [d005844](https://github.com/python-discord/bot/pull/2326/commits/d00584408a20e231df1e3d6ce3c4837e6bb829e4)

<img width="885" alt="image" src="https://user-images.githubusercontent.com/13956642/207570006-3f353a6e-ce84-4436-b7c7-9f3ceb7c2c88.png">

Reply:
Thanks for looking into it.